lib/chef/resource/windows_task.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Author:: Nimisha Sharad (<nimisha.sharad@msystechnologies.com>)
+# Copyright:: Copyright 2008-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class WindowsTask < Chef::Resource
+
+      provides :windows_task, os: "windows"
+
+      allowed_actions :create, :delete, :run, :end, :change, :enable, :disable
+      default_action :create
+
+      def initialize(name, run_context = nil)
+        super
+        @resource_name = :windows_task
+        @task_name = name
+        @action = :create
+      end
+
+      property :task_name, String, regex: [/\A[^\/\:\*\?\<\>\|]+\z/]
+      property :command, String
+      property :cwd, String
+      property :user, String, default: 'SYSTEM'
+      property :password, String
+      property :run_level, equal_to: [:highest, :limited], default: :limited
+      property :force, [TrueClass, FalseClass], default: false
+      property :interactive_enabled, [TrueClass, FalseClass], default: false
+      property :frequency_modifier, [Integer, String], default: 1
+      property :frequency, equal_to: [:minute,
+                                       :hourly,
+                                       :daily,
+                                       :weekly,
+                                       :monthly,
+                                       :once,
+                                       :on_logon,
+                                       :onstart,
+                                       :on_idle], default: :hourly
+      property :start_day, String
+      property :start_time, String
+      property :day, [String, Integer]
+      property :months, String
+      property :idle_time, Integer
+      property :random_delay, String

```

Comment:
@mwrock, when we accept input from the user in a different format than what is expected by the task, more parsing is required to ensure idempotency during `:create` action. It's doable but requires time.
I was working on https://github.com/chef-cookbooks/windows/issues/366. It's tricky to fix this because we are setting the properties of `@current_resource` using method `load_task_hash` to ensure idempotency: https://github.com/chef/chef/pull/5886/files#diff-208e6e972fb102d2fce9f211de0d94a8R262.
This method uses `schtasks /Query /FO LIST /V /TN <task_name>` to get the details of the task and doesn't return all the required properties. For some properties we need to inspect the XML object e.g.  `random_delay`. A lot of parsing is also required since the values that we get from the command and from the XML are not in the same format that the resource expects.
e.g. the resource expects `frequency` as let's say 15 and `frequency_modifier` as `minutes`, while the task XML will return `PT15M`.
A quick fix will be to update the task everytime `:create` action is called, but then we'll be compromising on the idempotency.